terraform-aws-modules/terraform-aws-rds-aurora
# chore: add all attributes and some outputs for enhanced monitoring IAM role

## Description
- add all attributes and some outputs for enhanced monitoring IAM role
- update `autoscaling` example for changes made here
- update variable descriptions (semantic corrects)

## Motivation and Context
- expose all attributes for configuring the enhanced monitoring IAM role

Closes #93
Closes #153 
Closes #196

## Breaking Changes
- Yes - users will need to make the following changes to

```hcl
	# Remove the following parameter from the module configuration
	# Functionality is stil maintained and provided by a data source so this value no longer needs to be set 
	# ref: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/partition)
	iam_partition = "aws-gov"
	
	# Change `permissions_boundary` to `iam_role_permissions_boundary`
	permissions_boundary = "arn:aws:..." -> iam_role_permissions_boundary = "arn:aws:..."
```

## How Has This Been Tested?
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects
- Test/verification stesp:
	- Deploy `examples/autoscaling` as currently defined on `master` branch
	- Switch to this branch and comment out lines 57-61 of `examples/autoscaling` (+outputs - this is 1:1 compared to what is currently on `master`)
	- Run `terraform plan` -> no changes detected
	- Uncomment lines 57-61 and outputs
	- Run `terraform apply`
	- Verify changes match desired outcome
